### ISSUE TITLE
Viewing the file in HDFViewer and Reading the data inside the code

### ISSUE THREAD

**Issue by SBahre** [reyntjesr/Hdf5DotnetTools#7]
I have been relying on the HDFViewer from the very beginning of using HDF5 files. Although the test cases of our repository are successful, I am not able to view the HDF5 files with compound dataset inside them.

This is the complete description of the issue.

I am writing a dataset using WriteCompounds in my HDF5 file `testCompounds.H5`
See the content of the struct identifier while **writing the dataset**
![image](https://user-images.githubusercontent.com/33088399/34563023-d8bcc60c-f176-11e7-84e6-35e13d2c2be7.png)

Notice the first two members: label and location.

See again the content while **reading the dataset**
![image](https://user-images.githubusercontent.com/33088399/34563203-a8e80a26-f177-11e7-80b1-e94ca2968790.png)

Finally the screenshot from the viewer.
![image](https://user-images.githubusercontent.com/33088399/34563328-3d6a224c-f178-11e7-89ad-2a074263cdfa.png)

--------------------------------------------------------------------------------------------------------------
Second test case results
WriteCompounds Case 2
![image](https://user-images.githubusercontent.com/33088399/34563483-d7f7b856-f178-11e7-8971-855b6ff8a0d6.png)

**Notice Time and location**
ReadCompunds Case 2
![image](https://user-images.githubusercontent.com/33088399/34563577-1b09ed30-f179-11e7-91e0-f2358e195a14.png)

HDFViewer Case 2
![image](https://user-images.githubusercontent.com/33088399/34563708-b1c46886-f179-11e7-8f9a-6878b1c95ccc.png)

In Summary,

1. In case 1, two String type values are being written and read in the program but are not visible in HDFviewer. Their datatypes are incorrect. also 
2. Time object, in Case 2, is not visible.
3. In case 2, String is not fully visible and dimensions are also incorrectly shown in the viewer.

String type is creating some problem in the viewer. String type is being handled differently in the viewer. Please review these doubts and help me correct the possible source of the bug in the whole scenario.

Attaching both the HDF5 files
[Debug.zip](https://github.com/reyntjesr/Hdf5DotnetTools/files/1603340/Debug.zip)

A quick reply would be appreciated.
